test/delayed_queue_test.rb

```diff
@@ -107,6 +107,30 @@
     Resque.enqueue_at(t, SomeIvarJob)
   end
 
+  test "multiple processes can safely handle delayed items" do
+    num_jobs = 5000
+    num_jobs.times do |i|
+      Resque.enqueue_at(Time.now + (i + 1), SomeIvarJob, job_number: i)
+    end
+
+    processes = 50
+    pids = []
+    jobs_enqueued = []
+
+    processes.times do
+      pids << fork_with_marshalled_pipe_and_result do
+        Resque::Scheduler.handle_delayed_items(Time.now + (num_jobs + 1))
+      end
+    end
+    jobs_enqueued += get_results_from_children(pids)
+
+    assert jobs_enqueued.all? { |enqueued| enqueued > 0 }
+    assert_equal num_jobs, Resque.redis.llen('queue:ivar')
+
+    actual_job_numbers = Resque.redis.lrange('queue:ivar', 0, -1).map { |job| Resque.decode(job)['args'][0]['job_number'] }.sort
+    assert_equal (0...num_jobs).to_a, actual_job_numbers
+  end
```

Comment:
Any ideas for additional race condition tests?

Reply:
Does this test actually fail without the zpop lua?

Reply:
yup 
```
Failure: test_multiple_processes_can_safely_handle_delayed_items(DelayedQueue)
./test/delayed_queue_test.rb:128:in `block (2 levels) in <main>'
<5000> expected but was
<18830>
```